### PR TITLE
Video orientation bugfix and test

### PR DESCRIPTION
**Pull request by nekronos**
Partially fixes issue https://github.com/fusetools/fuselibs-public/issues/1080.

This PR contains:
- [x] Changelog
- [ ] Documentation
- [x] Tests

### CODE REVIEW

File: Source/Fuse.Controls.Video/Android/VideoPlayer.uno [hunk 1]
```diff
@@ -196,25 +197,59 @@ namespace Fuse.Controls.VideoImpl.Android
...
+				try
+				{
+					android.media.MediaMetadataRetriever mmr = new android.media.MediaMetadataRetriever();
+					mmr.setDataSource(dataSorucePath);
+					String rotation = mmr.extractMetadata(android.media.MediaMetadataRetriever.METADATA_KEY_VIDEO_ROTATION);
+					if (rotation != null) {
+						return java.lang.Integer.parseInt(rotation);
+					}
+				}
+				catch(Exception e) {}
```
Comment (mortoray) [inline Source/Fuse.Controls.Video/Android/VideoPlayer.uno]: I'd prefer to see either a diagnostic here, or a comment indicating why we could just ignore the exception.

File: Source/Fuse.Controls.Video/Android/VideoPlayer.uno [hunk 2]
```diff
@@ -196,25 +197,59 @@ namespace Fuse.Controls.VideoImpl.Android
...
+						.openFd(dataSorucePath);
+
+					android.media.MediaMetadataRetriever mmr = new android.media.MediaMetadataRetriever();
+					mmr.setDataSource(afd.getFileDescriptor(), afd.getStartOffset(), afd.getLength());
+					String rotation = mmr.extractMetadata(android.media.MediaMetadataRetriever.METADATA_KEY_VIDEO_ROTATION);
+					if (rotation != null) {
+						return java.lang.Integer.parseInt(rotation);
+					}
+				}
+				catch (Exception e) {}
```
Comment (mortoray) [inline Source/Fuse.Controls.Video/Android/VideoPlayer.uno]: Also here.

File: Source/Fuse.Controls.Video/iOS/VideoImpl.mm [hunk 3]
```diff
@@ -294,6 +294,10 @@ int getRotation(void * videoState)
 				auto t = track.preferredTransform;
 				auto angle = atan2((double)t.b, (double)t.a);
 				auto d = angle * 180.0 / M_PI;
+				if (d < 0)
+				{
+					d = 360.0 + d;
```
Comment (mortoray) [inline Source/Fuse.Controls.Video/iOS/VideoImpl.mm]: This appears wrong. It uses 360degrees, but `d` is currently in radians `auto d = angle * 180.0 / M_PI;`

Comment (nekronos) [reply Source/Fuse.Controls.Video/iOS/VideoImpl.mm]: @mortoray what? `d` is degrees, read line `296` again

Comment (mortoray) [reply Source/Fuse.Controls.Video/iOS/VideoImpl.mm]: Derp! :face_with_head_bandage:  Yes, indeed. It's all degrees.

**Comment by nekronos** [comment]
@mortoray  fixed

**Comment by devadiab** [comment]
Thanks @nekronos and @mortoray, i'd like to be assured that you fixed the `CameraView `as well !

we need to add this line to `RecordingSession` Class 
```
        _mediaRecorder.setOrientationHint(Rotation);

```

**Comment by devadiab** [comment]
While the Rotation is the current rotation for the device

**Comment by nekronos** [comment]
@devadiab it has been fixed in `CameraView` aswell

**Comment by devadiab** [comment]
Thanks @nekronos

**Comment by devadiab** [comment]
Hi All,
This fix is not included in 1.8 (QA) version. It's too necessary for our launch

**Pull request merged**